pkg/schemacmp/lattice.go

```diff
@@ -278,6 +289,136 @@ func (a Byte) Join(other Lattice) (Lattice, error) {
 	}
 }
 
+// fieldTp is a mysql column field type implementing lattice.
+// It is used for the column field type (`github.com/pingcap/parser/types.FieldType.Tp`).
+type fieldTp struct {
+	value byte
+}
+
+// FieldTp is used for the column field type (`github.com/pingcap/parser/types.FieldType.Tp`).
+func FieldTp(value byte) Lattice {
+	return fieldTp{value: value}
+}
+
+// Unwrap implements Lattice
+func (a fieldTp) Unwrap() interface{} {
+	return a.value
+}
+
+// Compare implements Lattice.
+func (a fieldTp) Compare(other Lattice) (int, error) {
+	b, ok := other.(fieldTp)
+	if !ok {
+		return 0, typeMismatchError(a, other)
+	}
+
+	if a.value == b.value {
+		return 0, nil
+	}
+
+	// TODO: add more comparable type check here.
+	// maybe we can ref https://github.com/pingcap/tidb/blob/38f4d869d86c3b274e7d1998a52243a30b125c80/types/field_type.go#L325 later.
+	if mysql.IsIntegerType(a.value) && mysql.IsIntegerType(b.value) {
+		// special handle for integer type.
+		// TypeTiny(1) < TypeShort(2) < TypeInt24(9) < TypeLong(3) < TypeLonglong(8)
+		switch {
+		case a.value == mysql.TypeInt24:
+			if b.value <= mysql.TypeShort {
+				return 1, nil
+			}
+			return -1, nil
+		case b.value == mysql.TypeInt24:
+			if a.value <= mysql.TypeShort {
+				return -1, nil
+			}
+			return 1, nil
+		case a.value < b.value:
+			return -1, nil
+		default:
+			return 1, nil
+		}
```

Comment:
can we define a function to compare Integer type? and this function can be reused in `Join`

Reply:
addressed in https://github.com/pingcap/tidb-tools/pull/338/commits/fc42099d797032a437a22ce9f614f30af98a8106.